mantidproject/mantid
# Dis/enable beam centre component boxes on component update

**Description of work.**
This PR automatically updates the HAB/LAB component checkboxes in the ISIS SANS beam centre finder tab if the reduction mode has been changed.

If HAB mode has been selected, the beam centre finder LAB checkbox is unchecked and disabled, and vice versa. This change clarifies which components have been selected for reduction.

**Report to:** Steve King ISIS

**To test:**
1. Interfaces -> SANS -> ISIS SANS
2. In the settings tab, select the HAB reduction mode ("Hab" or "front")
3. Go to the beam centre finder tab. The HAB/Hab/front checkbox should be enabled and checked. The LAB/main-detector/rear checkbox should be unchecked and disabled
4. Back in settings, select LAB reduction mode ("main-detector" or "rear")
5. The checkboxes should be in the opposite state to step (3)
6. Back in settings, select "merged" or "all" mode
7. Both checkboxes in beam center finder tab should be enabled and checked

Fixes #25997 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
